go-controller/pkg/ovn/policy.go

```diff
@@ -0,0 +1,532 @@
+			match := fmt.Sprintf("match=\"%s && %s\"", l3Match, outputMatch)
+
+			oc.addAclAllow(policy.Namespace, policy.Name,
+				pod.Spec.NodeName, logical_port, match, i)
+		}
+		for _, port := range ingress.portPolicies {
+			l4Match := ""
+			if port.protocol == "TCP" {
+				l4Match = fmt.Sprintf("tcp && tcp.dst==%d",
+					port.port)
+			} else if port.protocol == "UDP" {
+				l4Match = fmt.Sprintf("udp && udp.dst==%d",
+					port.port)
+			} else {
+				continue
+			}
+			match := fmt.Sprintf("match=\"%s && %s && %s\"",
+				l3Match, l4Match, outputMatch)
+			oc.addAclAllow(policy.Namespace, policy.Name,
+				pod.Spec.NodeName, logical_port, match, i)
+		}
+	}
+}
+
+func (oc *Controller) handleLocalPodSelectorDelFunc(
+	policy *kapisnetworking.NetworkPolicy, np *namespacePolicy,
+	obj interface{}) {
+	pod := obj.(*kapi.Pod)
+
+	if pod.Spec.NodeName == "" {
+		return
+	}
+
+	// Get the logical port name.
+	logical_port := fmt.Sprintf("%s_%s", pod.Namespace, pod.Name)
+
+	// For each ingress rule, remove the ACL
+	for i, ingress := range np.ingressPolicies {
+		l3Match := fmt.Sprintf("ip4.src == {$%s}", ingress.localPeerPods)
+		outputMatch := fmt.Sprintf("outport == \\\"%s\\\"", logical_port)
+		if len(ingress.portPolicies) == 0 {
+			match := fmt.Sprintf("match=\"%s && %s\"", l3Match, outputMatch)
+			oc.deleteAclAllow(policy.Namespace, policy.Name,
+				pod.Spec.NodeName, logical_port, match, i)
+		}
+		for _, port := range ingress.portPolicies {
+			l4Match := ""
+			if port.protocol == "TCP" {
+				l4Match = fmt.Sprintf("tcp && tcp.dst==%d",
+					port.port)
+			} else if port.protocol == "UDP" {
+				l4Match = fmt.Sprintf("udp && udp.dst==%d",
+					port.port)
+			} else {
+				continue
+			}
+			match := fmt.Sprintf("match=\"%s && %s && %s\"",
+				l3Match, l4Match, outputMatch)
+			oc.deleteAclAllow(policy.Namespace, policy.Name,
+				pod.Spec.NodeName, logical_port, match, i)
+		}
+	}
+}
+
+func (oc *Controller) handleLocalPodSelector(
+	policy *kapisnetworking.NetworkPolicy, np *namespacePolicy) {
+
+	client, _ := oc.Kube.(*kube.Kube)
+	clientset, _ := client.KClient.(*kubernetes.Clientset)
+	podSelectorAsSelector, _ := metav1.LabelSelectorAsSelector(
+		&policy.Spec.PodSelector)
+
+	watchlist := newListWatchFromClient(clientset.Core().RESTClient(), "pods",
+		policy.Namespace, podSelectorAsSelector)
+
+	_, controller := cache.NewInformer(
+		watchlist,
+		&v1.Pod{},
+		time.Second*0,
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				oc.handleLocalPodSelectorAddFunc(policy, np, obj)
+			},
+			DeleteFunc: func(obj interface{}) {
+				oc.handleLocalPodSelectorDelFunc(policy, np, obj)
+			},
+			UpdateFunc: func(oldObj, newObj interface{}) {
+				oc.handleLocalPodSelectorAddFunc(policy, np, newObj)
+			},
+		},
+	)
+	channel := make(chan struct{})
+	go controller.Run(channel)
+	select {
+	case <-np.stop:
+		channel <- struct{}{}
+	}
+}
+
+func (oc *Controller) handlePeerPodSelector(
```

Comment:
I think that was the idea of the shared informer. Let me think and experiment with the common shared informer factory code and see if we can apply it here. Then, we may just have one watcher, but multiple consumers based on the filters that they use.
Let's do it in v2 again.